compiler/rustc_hir_typeck/src/fn_ctxt/checks.rs

```diff
@@ -157,8 +160,8 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         c_variadic: bool,
         // Whether the arguments have been bundled in a tuple (ex: closures)
         tuple_arguments: TupleArgumentsFlag,
-        // The DefId for the function being called, for better error messages
-        fn_def_id: Option<DefId>,
+        // The callee type (e.g. function item ZST, function pointer or closure).
```

Comment:
It may be useful to note that this will be the `FnDef` for `FnOnce::call_once` if, for example, we try to call a `Box<dyn FnOnce()>`, etc.